api/repositories/service_offering_repository_test.go

```diff
@@ -317,4 +330,136 @@ var _ = Describe("ServiceOfferingRepo", func() {
+
+			Expect(k8sClient.Update(ctx, instance)).To(Succeed())
+
+			binding = &korifiv1alpha1.CFServiceBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      uuid.NewString(),
+					Namespace: space.Name,
+					Labels: map[string]string{
+						korifiv1alpha1.PlanGUIDLabelKey: plan.Name,
+					},
+				},
+				Spec: korifiv1alpha1.CFServiceBindingSpec{
+					Service: corev1.ObjectReference{
+						Kind:       "CFServiceInstance",
+						APIVersion: korifiv1alpha1.SchemeGroupVersion.Identifier(),
+						Name:       instance.Name,
+					},
+					AppRef: corev1.LocalObjectReference{
+						Name: "some-app-guid",
+					},
+				},
+			}
+
+			binding.Finalizers = append(binding.Finalizers, korifiv1alpha1.CFServiceBindingFinalizerName)
+			Expect(k8sClient.Create(ctx, binding)).To(Succeed())
+
+			message = repositories.DeleteServiceOfferingMessage{GUID: offering.Name}
+		})
+
+		JustBeforeEach(func() {
+			createRoleBinding(ctx, userName, spaceDeveloperRole.Name, space.Name)
+			deleteErr = repo.DeleteOffering(ctx, authInfo, message)
+		})
+
+		It("successfully deletes the offering", func() {
+			Expect(deleteErr).ToNot(HaveOccurred())
+
+			namespacedName := types.NamespacedName{
+				Name:      offering.Name,
+				Namespace: rootNamespace,
+			}
+
+			err := k8sClient.Get(context.Background(), namespacedName, &korifiv1alpha1.CFServiceOffering{})
+			Expect(k8serrors.IsNotFound(err)).To(BeTrue(), fmt.Sprintf("error: %+v", err))
+		})
+
+		When("the service offering does not exist", func() {
+			BeforeEach(func() {
+				message.GUID = "does-not-exist"
+			})
+
+			It("returns a error", func() {
+				Expect(errors.As(deleteErr, &apierrors.NotFoundError{})).To(BeTrue())
+			})
+		})
+
+		When("Purge is set to true", func() {
+			BeforeEach(func() {
+				message.Purge = true
+			})
+			It("successfully deletes the offering and all related resources", func() {
+				Expect(deleteErr).ToNot(HaveOccurred())
+
+				err := k8sClient.Get(context.Background(), types.NamespacedName{Name: offering.Name, Namespace: rootNamespace}, &korifiv1alpha1.CFServiceOffering{})
+				Expect(k8serrors.IsNotFound(err)).To(BeTrue(), fmt.Sprintf("error: %+v", err))
+
+				err = k8sClient.Get(context.Background(), types.NamespacedName{Name: plan.Name, Namespace: rootNamespace}, &korifiv1alpha1.CFServicePlan{})
+				Expect(k8serrors.IsNotFound(err)).To(BeTrue(), fmt.Sprintf("error: %+v", err))
+
+				err = k8sClient.Get(context.Background(), types.NamespacedName{Name: instance.Name, Namespace: space.Name}, &korifiv1alpha1.CFServiceInstance{})
+				Expect(k8serrors.IsNotFound(err)).To(BeTrue(), fmt.Sprintf("error: %+v", err))
+
+				serviceBinding := new(korifiv1alpha1.CFServiceBinding)
+				err = k8sClient.Get(context.Background(), types.NamespacedName{Name: binding.Name, Namespace: space.Name}, serviceBinding)
+
+				Expect(err).ToNot(HaveOccurred())
+				Expect(serviceBinding.Finalizers).To(BeEmpty())
```

Comment:
Here the service binding won't be deleted, but we can still check if the finalizer is removed, which is what we actually do in the repo.